src/main/java/io/takari/aether/connector/AetherRepositoryConnectorFactory.java

```diff
@@ -59,34 +68,113 @@ public boolean equals(Object obj) {
     }
   }
 
-  // Default constructor required for the service locator to work with you use this factory outside the confines of Guice.
-  public  AetherRepositoryConnectorFactory() throws NoSuchAlgorithmException {
-    this(null, null);
+  // Default constructor required for the service locator to work with you use
+  // this factory outside the confines of Guice.
+  public AetherRepositoryConnectorFactory() throws NoSuchAlgorithmException {
+    this(null, null, null);
   }
-  
+
   @Inject
-  public AetherRepositoryConnectorFactory(FileProcessor fileProcessor, @Nullable SSLSocketFactory sslSocketFactory) throws NoSuchAlgorithmException {
+  public AetherRepositoryConnectorFactory(FileProcessor fileProcessor, @Nullable SSLSocketFactory sslSocketFactory,
+      @Nullable X509TrustManager trustManager) throws NoSuchAlgorithmException {
     this.fileProcessor = fileProcessor;
-    
-    // explicitly use jdk-default ssl socket factory if none is provided externally
+
+    // explicitly create new ssl socket factory if none is provided externally
     // this is necessary because okhttp default socket factory does not honour
-    // javax.net.ssl.keyStore/trustStore system properties, which are the only way
+    // javax.net.ssl.keyStore/trustStore system properties, which are the only
+    // way
     // to use custom ket/trust stores in maven in m2e
-    this.sslSocketFactory = sslSocketFactory != null? sslSocketFactory: SSLContext.getDefault().getSocketFactory();
+    if (sslSocketFactory != null && trustManager != null) {
+      this.sslSocketFactory = sslSocketFactory;
+      this.trustManager = trustManager;
+    } else {
+      // can not use use jdk-default ssl socket factory like previous because
+      // we can not get trust manager without reflection. So create new ssl
+      // socket factory using system properties to create fake jdk-default
+      // ssl socket factory.
+      try {
+        String trustManagerFactoryAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+        // Clearly specify TrustManagerFactory provider to SunJSSE.
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(trustManagerFactoryAlgorithm, "SunJSSE");
```

Comment:
This apparently is not compatible with [IBM JDK](https://bugs.eclipse.org/bugs/show_bug.cgi?id=515859#c18)